File: grokking algo/binarySearch.py
def binary_search(list, item):
    low = 0
    high = len(list) - 1

    while low <= high:
        mid = (low + high) / 2
        mid = round(mid)
        guess = list[mid]
        if guess == item:
            return mid
        if guess > item:
            high = mid - 1
        else:
            low = mid + 1
    return None
    
my_list = [1,2,4,5,7,9]

print(binary_search(my_list, 1))
print(binary_search(my_list, 2))
print(binary_search(my_list, 3))
print(binary_search(my_list, 9))

File: visualStudio/binarySearch/BinarySearch.java
public class BinarySearch {
    public static void main(String[] args) {
        // Crea un array de enteros ordenados
        int[] array = {1, 2, 3, 4, 5};

        // Llamamos a la función de búsqueda binaria y buscamos el número 3
        int index = binarySearch(array, 3);

        // Si el número se encontró, mostramos su índice
        if (index != -1) {
            System.out.println("El número se encontró en el índice " + index);
        } else {
            System.out.println("El número no se encontró en el array");
        }
    }

    public static int binarySearch(int[] array, int item) {
        int low = 0;
        int high = array.length - 1;

        while (low <= high) {
            int middle = (low + high) / 2;
            if (array[middle] == item) {
                return middle;
            } else if (array[middle] < item) {
                low = middle + 1;
            } else {
                high = middle - 1;
            }
        }

        return -1;
    }
}